pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -779,7 +779,7 @@ public boolean hasMessageAvailable() throws PulsarClientException {
             if (exception != null) {
                 completableFuture.completeExceptionally(exception);
             } else {
-                completableFuture.complete(hasMessageAvailable.get());
+                completableFuture.complete(hasMessageAvailable.get() || numMessagesInQueue() > 0);
```

Comment:
What about moving the check on the number of messages on the top of the method?
If we have messages we can skip doing the other things.

Bu the way I am not sure if doing so would alter the way it works significantly, as in case of having messages we won't call hasMessageAvailableAsync on the wrapped  consumers

Did you think about this possibility?

Reply:
You're right. I think moving to the top of the method doesn't change the behavior. 

In the previous implementation, even if `hasMessageAvailableAsync` is true, the `readNext()` still might blocking.